util/controller.go

```diff
@@ -0,0 +1,74 @@
+package util
```

Comment:
Maybe we should rename package to 'base' or 'common' since it's less of a util package now.